### PR TITLE
Fixing UI Installation

### PR DESCRIPTION
**Pull request by miroswan**
I believe /var/lib/consul needs to exist in order to symlink /opt/consul-webui/$VERSION to /var/lib/consul/ui. This was failing for me prior to adding the directory resource to the LWRP for web-ui installation.

### CODE REVIEW

**Comment by miroswan** [comment]
Looks like the tests need to be fixed themselves.

**Comment by Ginja** [comment]
Thanks for the PR! A few requests:

* Could you change the directory resource so it's not hardcoded, but rather reflects where it's being symlinked (see below)?
* Could you add a resource to remove the directory in `action_remove`?
* If you wouldn't mind taking a shot at fixing that'd be awesome, but don't sweat it.

```ruby

base = ::File.dirname options[: symlink_to]
# => /var/lib/consul

directory base
```

**Comment by miroswan** [comment]
I can take a shot at this later in the week.

**Comment by miroswan** [comment]
I fixed my tests locally and updated travis.yml in attempt to fix the faraday situation. Don't think travis is configured to run subsequent builds against this PR so it's hard to tell if my changes will fix the faraday issue as well.

**Comment by miroswan** [comment]
It appears that faraday 0.9.0 is having problems. When I bump it down to 0.8.0, the problem goes away, but then poise-boiler reverts to version 0. Seems like a strange dependency issue.

File: Gemfile [hunk 1]
```diff
@@ -25,3 +25,9 @@ end
 group :doc do
   gem 'yard'
 end
+
+# Referencing https://github.com/nomad/shenzhen/issues/96 to
+# fix build issue regarding the faraday gem
+group :faraday do
+  gem 'faraday', '~> 0.8.0'
```
Comment (shortdudey123) [inline Gemfile]: This is due to a bug in Berkshelf 3.x that was fixed in 4.x

**Comment by shortdudey123** [comment]
Tests are correctly fixed in https://github.com/johnbellone/consul-cookbook/pull/305

**Comment by shortdudey123** [comment]
@miroswan can you rebase on master? several tests have been fixed causing your merge conflicts

**Comment by miroswan** [comment]
Hadn't checked in on this in a while. I'll take a look at this soon.

**Comment by miroswan** [comment]
I rebased as requested, but it appears that some previous commits from upstream (for which I am not the author) have snuck into my PR. Outside of that, the tests are passing.

**Comment by miroswan** [comment]
I think I'm going to do a fresh PR since it's a small change.

**Comment by miroswan** [comment]
It appears that someone added my fix from a different branch. Just waiting for the next release to the supermarket now.

**Comment by gdavison** [comment]
Yeah, I hadn't noticed you'd submitted a PR when I created mine. If you had anything in your PR that you think should be added, go ahead.

**Pull request closed without merging**